src/Credentials/GCECredentials.php

```diff
@@ -105,8 +110,11 @@ public static function onGce(ClientInterface $client = null)
       // could lead to false negatives in the event that we are on GCE, but
       // the metadata resolution was particularly slow. The latter case is
       // "unlikely".
-      $resp = $client->get($checkUri, ['timeout' => 0.3]);
-      return $resp->getHeader(self::FLAVOR_HEADER) == 'Google';
+      $resp = $httpHandler(
+        new Request('GET', $checkUri),
+        ['timeout' => 0.3]
```

Comment:
That's a great question.

When using guzzle the options are:
http://docs.guzzlephp.org/en/5.3/clients.html#request-options
http://docs.guzzlephp.org/en/latest/request-options.html

But that doesn't really help someone who defines their own implementation. Maybe we should come up with a whitelist of valid options or at the least document the options that are required to be implemented such as `timeout` above?